### PR TITLE
add js version pre-commit

### PR DESCRIPTION
**Pull request by gengjiawen**
I find it very convenient and very easy to config.

### CODE REVIEW

File: docs/precommit.md [hunk 1]
```diff
@@ -30,7 +30,27 @@ and add this config to your `package.json`:
...
+
+```json
+{
+  "scripts": {
+    "prettier": "prettier */**/*.js --ignore-path ./.prettierignore --write && git add . && git status"
+   },
+   "pre-commit": [
+    "prettier",
+  ]
+}
```
Comment (j-f1) [inline docs/precommit.md]: Can you put a closing ` ``` ` here?

Comment (gengjiawen) [reply docs/precommit.md]: silly me, forget that :)

**Comment by gengjiawen** [comment]
Have not got a clue how to fix the ci ...

**Comment by duailibe** [comment]
Thank you!

re CI: prettier was changing the contents of the `package.json` file, but your version was better so I just added `prettier-ignore` there :-)

**Comment by azz** [comment]
Is this a good idea? This will not respect the files staged in git and commit everything. Better to recommend lint-staged approach.

**Comment by gengjiawen** [comment]
yes, maybe should use something like pipe to redirect to the file list to git add.

**Comment by gengjiawen** [comment]
I find someone use `lint-stage` with `pre-commit`, like https://github.com/apollographql/apollo-link/blob/bda4e08ae3baf53a4b90987e9b43766474e7ad78/package.json#L86. Maybe this is a better way.

**Pull request merged**